notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/JpaNotificationService.java

```
@@ -94,7 +95,26 @@ public NotificationResponse fetch(PortletRequest req) {
         return rslt;
 
     }
+	
+    public void updateEntryState(PortletRequest req, String entryId, NotificationState state) {
+        if (usernameFinder.isAuthenticated(req)) {
+            final String username = usernameFinder.findUsername(req);
+
+            String idStr = entryId.replaceAll(ID_PREFIX, ""); // remove the prefix
+
+            JpaEntry jpaEntry = notificationDao.getEntry(Long.parseLong(idStr));
+            if (jpaEntry != null) {
+                JpaEvent event = new JpaEvent();
+                event.setEntry(jpaEntry);
+                event.setState(state);
+                event.setTimestamp(new Timestamp(new Date().getTime()));
+                event.setUsername(username);
 
+                notificationDao.createOrUpdateEvent(event);
+            }
+        }
```

Should probably throw an IllegalArgumentException if the entry is not found.